identity/backend/src/main/java/io/camunda/identity/usermanagement/repository/MembershipRepository.java

```diff
@@ -7,28 +7,21 @@
  */
 package io.camunda.identity.usermanagement.repository;
 
-import io.camunda.identity.usermanagement.Group;
+import io.camunda.identity.usermanagement.CamundaGroup;
+import io.camunda.identity.usermanagement.model.Membership;
 import java.util.List;
-import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public class MembershipRepository {
-  public static final String DEF_USERS_GROUPS_QUERY =
-      "select g.id, g.group_name"
-          + " from groups g, group_members gm"
-          + " where gm.username = ? and g.id = gm.group_id";
+public interface MembershipRepository extends JpaRepository<Membership, Long> {
 
-  private final JdbcTemplate jdbcTemplate;
-
-  public MembershipRepository(final JdbcTemplate jdbcTemplate) {
-    this.jdbcTemplate = jdbcTemplate;
-  }
-
-  public List<Group> loadUserGroups(final String username) {
-    return jdbcTemplate.query(
-        DEF_USERS_GROUPS_QUERY,
-        new String[] {username},
-        (rs, rowNum) -> new Group(rs.getInt(1), rs.getString(2)));
-  }
+  @Query(
+      value =
+          "select new io.camunda.identity.usermanagement.CamundaGroup(gm.group.id, gm.group.name)"
```

Comment:
Would it be possible to just use `CamundaGroup` without the full reference?

Reply:
I tried to drop the package name but it was not working, if you know how to do it or have any idea about it, please share. :)

Reply:
I wonder if it would auto-map it by dropping the new statement altogether ... like

`select gm.group.id, gm.group.name...` 

have you tried that by any chance? 🤔 🤔 🤔

Reply:
EDIT: maybe by giving the fields alias names that match the CamundaUser fields id and name ? 🤔

Reply:
unfortunately this approach only works with interface return type, not class/record.

Reply:
I see, that's good to know, thanks for sharing :) ... well, it's no big deal imo, let's leave it as is, wdyt?

Reply:
We do have examples in the current Identity repo where we let the response be auto mapped to a class/record, for example https://github.com/camunda-cloud/identity/blob/main/management-api/src/main/java/io/camunda/identity/repository/ResourceRepository.java#L22-L32